### DISCUSSION TITLE
Report all trace data to SkyWalking manually

### DISCUSSION THREAD

**Discussion by leopardrota** [apache/skywalking#10835] [Q&A] [answered]
I try to learn how to report all trace data to SkyWalking manually in this [doc](https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/application-toolkit-trace/). Now I see how to build custom trace data, but I do not yet know how to connect and report to SkyWalking OAP after reading.
Is there a complete demo, guidance or something like that somewhere? If no, can someone tell me how to report the trace data after being prepared? Also, I want the trace data reporting process to be exclusively manual and no reporting from other places.
Thanks!

**Comment by wu-sheng** [comment, marked as answer]
That is a toolkit and bridge only. You still need Agent to report the data.
We don't have pure SDK for the data collecting.

**Reply by leopardrota** [reply]
Oh, I see. It means, in addition to the data I manually report, there will also be the data collected and reported by Agent.

**Reply by wu-sheng** [reply]
Yes. It is correct.

**Reply by Superskyyy** [reply]
> Oh, I see. It means, in addition to the data I manually report, there will also be the data collected and reported by Agent.

you can always disable agent plugins according to your own needs, just reusing the reporting core.
